R/corrpart.b.R

```diff
@@ -246,72 +233,59 @@ corrPartClass <- R6::R6Class(
+	                    if (det(cvx) < .Machine$double.eps) { icvx <- ginv(cvx) } else { icvx <- solve(cvx) }
+	                
+	                    # calculation of the partial correlation coefficient 
+                        if (type == 'part') {
+                            results[ifelse(method == 'pearson', 'r', ifelse(method == 'spearman', 'rho', 'tau')), 1] <- -cov2cor(icvx)[1, 2]
+                        # calculation of the semi-partial correlation coefficients
+                        # this is more complex, to speed up processing both row and column variables
+                        # are calculated at once and given back in a list with two entries
+                        } else {
+                            spcor <- -cov2cor(icvx) / sqrt(diag(cvx)) / sqrt(abs(diag(icvx) - t(t(icvx ^ 2) / diag(icvx))))
+                            results[ifelse(method == 'pearson', 'r', ifelse(method == 'spearman', 'rho', 'tau')), 1] <- spcor[1, 2]
+                            results[ifelse(method == 'pearson', 'r', ifelse(method == 'spearman', 'rho', 'tau')), 2] <- spcor[2, 1]
+                        }
+                
+	                # assign p-value 
+                        # repeated depending on whether it is a semi-partial correlation (d = 2) or not (d = 1)
+                        for (i in 1:d) {            
+                            # calculate the p-values for undirected ('corr') and directed hypotheses ('pos', 'neg')
+                            # if the correlation coefficient doesn't match the proposed direction, the value is not set and remains NA
+                            if (hyp == 'corr') {
+                                if      (method == 'pearson')  { results[, i]$rp   <- 2 *    pt(-abs(results[, i]$r   * sqrt((n - 2 - c) / (1 - results[, i]$r   ^ 2))), (n - 2 - c)) }
+                                else if (method == 'spearman') { results[, i]$rhop <- 2 *    pt(-abs(results[, i]$rho * sqrt((n - 2 - c) / (1 - results[, i]$rho ^ 2))), (n - 2 - c)) }
+                                else if (method == 'kendall')  { results[, i]$taup <- 2 * pnorm(-abs(results[, i]$tau / sqrt(2 * (2 * (n - c) + 5) / (9 * (n - c) * (n - 1 - c)))))   }
+                            } else if ((hyp == 'neg' && results[, i]$r   < 0) || (hyp == 'pos' && results[, i]$r   > 0)) {
+                                if      (method == 'pearson')  { results[, i]$rp   <-        pt(-abs(results[, i]$r   * sqrt((n - 2 - c) / (1 - results[, i]$r   ^ 2))), (n - 2 - c)) }
+                                else if (method == 'spearman') { results[, i]$rhop <-        pt(-abs(results[, i]$rho * sqrt((n - 2 - c) / (1 - results[, i]$rho ^ 2))), (n - 2 - c)) }
+                                else if (method == 'kendall')  { results[, i]$taup <-     pnorm(-abs(results[, i]$tau / sqrt(2 * (2 * (n - c) + 5) / (9 * (n - c) * (n - 1 - c)))))   }
+                            }
+                        } # for
+                    }) # try
+                } # for
```

Comment:
These comments can be removed (this should be clear using a good IDE)

Reply:
Yes, maybe. Which IDE are you using?